lib/collection/src/shards/shard_holder/mod.rs

```diff
@@ -445,10 +445,12 @@ impl ShardHolder {
                 shard_key: resharding_state.shard_key.clone(),
                 comment: status.map(|p| p.comment),
             });
+        } else {
+            return None;
```

Comment:
I think you can use `?` in `let resharding_state = &*self.resharding_state.read()?;` and avoid returning `None` here

Reply:
Not really, because it's a parking_lot RwLockGuard that doesn't implement Try itself.

But I've moved the return up:

```rust
        let Some(resharding_state) = &*self.resharding_state.read() else {
            return None;
        };
```

Thanks!